docs/proposals/data-plane-config.md

```diff
@@ -1,21 +1,128 @@
 # Enhancement Proposal-929: Data Plane Dynamic Configuration
 
 - Issue: https://github.com/nginxinc/nginx-kubernetes-gateway/issues/929
-- Status: Provisional
+- Status: Implementable
 
 ## Summary
 
-This proposal is intended to contain the design for how to dynamically configure the data plane for the
-NGINX Gateway Fabric (NGF) project. Similar to control plane configuration, we should be able to leverage
+This proposal is intended to contain the design for how to dynamically configure global settings for the data plane
+of the NGINX Gateway Fabric (NGF) product. Similar to control plane configuration, we should be able to leverage
 a custom resource definition to define data plane configuration, considering fields such as telemetry and
 upstream zone size.
 
 ## Goals
 
-Define a CRD to dynamically configure various settings for the NGF data plane. The initial configurable options
-will be for telemetry (tracing) and upstream zone size.
+Define a CRD to dynamically configure various global settings for the NGF data plane. The initial configurable
+options will be for telemetry (tracing) and upstream zone size.
 
 ## Non-Goals
 
  1. This proposal is not defining every setting that needs to be present in the configuration.
  2. This proposal is not for any configuration related to control plane.
+
+## Introduction
+
+The NGF data plane will evolve to have various user-configurable options. These could include, but are not
+limited to, tracing, logging, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NGF. The first set of options that we will allow users to
+configure are tracing and upstream zone size. The easiest and most intuitive way to implement a Kubernetes-native
+API is through a CRD.
+
+The purpose of this CRD is to contain "global" configuration options for the data plane, and not focused on policy
+per route or backend.
+
+In this doc, the term "user" will refer to the cluster operator (the person who installs and manages NGF). The
+cluster operator owns this CRD resource.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxProxy
+metadata:
+    name: nginx-proxy-config
+    namespace: nginx-gateway
+spec:
+    telemetry:
+        exporters:
+            otlp:
+                endpoint: my-otel-collector.svc:4317
+                interval: 5s
+                batchSize: 512
+                batchCount: 4
+    upstreamZoneSize: 1024k
```

Comment:
I could have a top level `http` field for now.

```
spec:
    http:
      upstreamZoneSize: 512k
```

Reply:
@pleshakov Do you think `telemetry` should live under `http` since it only works in the http context? Should we have `http` and `stream`(future) as our top level fields, and everything else lives underneath? Or is `telemetry` general enough that it can just be at the top level?

Reply:
@sjberman 
telemetry seems general enough.
but since otel module specifically works for http module, having it under http makes sense to me.

Going forward, such a split (http and stream) will help us separate the config parts between HTTP related and TCP/UDP related.

Reply:
I think the only concern I have is if we support a metrics exporter config in the future while having both http and stream, it could result in a user having to define it twice, when they probably want both to go to the same place.

Reply:
@pleshakov What do you think of my last comment?

Reply:
@sjberman 
> I think the only concern I have is if we support a metrics exporter config in the future while having both http and stream, it could result in a user having to define it twice, when they probably want both to go to the same place.

I think it is a valid concern.
At the same time, If that happens, will be able to add a global one in receptiveness becomes an issue?
Also, I wonder if tracing is really a use case for TCP connections or UDP datagrams? (meaning are we going to have that problem at all)

In my view, having a clean separation between http and stream is more beneficial. 

Some similar examples:
The error log works for both http and stream, it will be global
However, access log is specific to either http or stream.